vortex-array/src/vtable/statistics.rs

```diff
@@ -26,58 +26,56 @@ where
 
 impl Array {
     /// Computes ths statistics for the given array and stat. This will update the stats of the array
-    /// and return this set.
+    /// and return this stats_set.
     ///
     /// Other stats might be computed or inferred at the same time.
     pub fn compute_statistics(&self, stat: Stat) -> VortexResult<StatsSet> {
         if self.is_empty() {
             return Ok(StatsSet::empty_array());
         }
 
-        if let Some(stat) = self.get(stat) {
+        if let Some(stat) = self.get_stat(stat) {
             if stat.is_exact() {
-                return Ok(self.to_set());
+                return Ok(self.stats_set());
             }
         }
 
-        let mut set = if stat == Stat::Min || stat == Stat::Max {
-            // min max sets the array stats
-            let _min_max = min_max(self)?;
-            let set = self.to_set();
-            if let Some(MinMaxResult { min, max }) = _min_max {
-                self.to_set().combine_sets(
```

Comment:
this is the bug, we mutate the array's set but return an older copy.

Reply:
This fix looks good.